docs/markdown/Writing Plugins/tutorials/01-create-new-goal-with-a-custom-target.md

```diff
@@ -0,0 +1,335 @@
+# Introduction
+
+In this tutorial, you'll learn the basics needed to get started writing a plugin. You will create a new goal, `project-version`, which will tell you the version (retrieved from the `VERSION` text file) of a particular project in your monorepository. You will learn how to create a new custom target to refer to the `VERSION` file, how to author a new goal, and, most importantly, how to connect rules and targets. You can follow along this tutorial in your own repository; you only need to be on a recent version of Pants and have a `VERSION` file containing a version string e.g. `1.2.3`.
+
+## Registering a plugin
+
+We'll be writing an [in-repo plugin](https://www.pantsbuild.org/docs/plugins-overview#in-repo-plugins), and expect you to have the `pants-plugins/project_version` directory with the `pants.toml` configuration file extended:
+
+```toml pants.toml
+pythonpath = ["%(buildroot)s/pants-plugins"]
```

Comment:
This is because this is an in-repo plugin that should run directly from sources, right?

Probably important to talk about the distinction between published plugins that you consume as a requirement, vs in-repo plugins consumed as source?

Reply:
Perhaps no need to go into detail, and the section open with "We'll be writing an in-repo plugin.." however a one line comment may serve well (esp. if this snippet is copy-pasted it hopefully follows along)

```toml
# Set `pythonpath` for the in-repo Pants plugin:
pythonpath = ...
```

Reply:
Thanks @kaos, done!